File: 0438-find-all-anagrams-in-a-string/0438-find-all-anagrams-in-a-string.cpp
class Solution {
public:
    vector<int> findAnagrams(string s, string p) {
        vector<int> Pcount(26,0);
        vector<int> Scount(26,0);
        vector<int>result;
        for(auto x : p){
            Pcount[x-'a']++;
        }
        for(int i=0;i<s.size();i++){
           Scount[s[i]-'a']++;
           if(i>=p.size())
           Scount[s[i - p.size()] - 'a']--;
           if(Pcount == Scount)
           result.push_back(i-p.size()+1);
        }

        return result;
        
    }
};

File: 0438-find-all-anagrams-in-a-string/0438-find-all-anagrams-in-a-string.py
class Solution:
    def findAnagrams(self, s: str, p: str) -> List[int]:
        pcount=[0]*26
        scount=[0]*26
        result=[]
        for i in  p:
            pcount[ord(i)-ord('a')]+=1
        for i in range(len(s)):
            scount[ord(s[i])-ord('a')]+=1
            if i>=len(p):
                 scount[ord(s[i-len(p)])-ord('a')]-=1
            if(pcount==scount):
                result.append(i-len(p)+1)
        return result
